flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -317,7 +337,7 @@ private CompletableFuture<Void> shutdownInternally(Duration timeout) {
             LOG.debug("Shutting down rest endpoint.");
 
             if (bootstrap != null) {
-                if (bootstrap.config().group() != null) {
+                if (bootstrap.config().group() != null && useInternalEventLoopGroup) {
```

Comment:
why does this if have `&& useInternalEventLoopGroup`? I would assume we want to pass the group through whether we have an internal group or whether one has been supplied.